libpod/image/image.go

```diff
@@ -1075,3 +1077,176 @@ func (i *Image) Comment(ctx context.Context, manifestType string) (string, error
 	}
 	return ociv1Img.History[0].Comment, nil
 }
+
+//LayerInfo keeps information of single layer
+//along with ID of its parent and Child layer.
+type LayerInfo struct {
+	ID       string
+	ParentID string
+	//There can be multiple childs (Fork)
+	ChildID  []string
+	RepoTags []string
+	Size     int64
+}
+
+//InfoImage keep information of Image along with all associated layers
+type InfoImage struct {
+	ID       string
+	RepoTags []string
+	Size     int64
+	Layers   []LayerInfo
+}
+
+// Tree gets dependency list in tree format of the image
+func (i *Image) Tree(ctx context.Context, imageName string, whatrequires bool) (*bytes.Buffer, error) {
+	layerInfoMap, err := buildImageLayersDependencyMap(ctx, i, whatrequires)
+	if err != nil {
+		return nil, err
+	}
+
+	imageTopLayer := i.TopLayer()
+	ll, ok := layerInfoMap[imageTopLayer]
+	if !ok {
+		return nil, fmt.Errorf("top-layer of image not found in dependency-map")
+	}
+
+	// Create a imageInfo and fill the image and layer info
+	imageInfo := &InfoImage{
+		ID:       i.ID(),
+		RepoTags: i.Names(),
+	}
+	var buffer bytes.Buffer
+
+	if !whatrequires {
+		err := buildImageDependencyMap(i, imageInfo, layerInfoMap, ll.ID)
+		if err != nil {
+			return nil, err
+		}
+		err = formatImageDependencyOutputBuffer(&buffer, imageInfo)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		err := buildImageDependentsMap(i, imageInfo, layerInfoMap, ll.ID, 0, &buffer)
+		if err != nil {
+			return nil, err
+		}
+
+		//Add image info at beginning of buffer.
+		treeData := buffer.String()
+		buffer.Reset()
+		buffer.WriteString(fmt.Sprintf("Image ID     : %s\n", imageInfo.ID[:12]))
+		buffer.WriteString(fmt.Sprintf("RepoTags     : %s\n", imageInfo.RepoTags))
+		buffer.WriteString(fmt.Sprintf("Size         : %v\n", units.HumanSizeWithPrecision(float64(imageInfo.Size), 4)))
+		buffer.WriteString(fmt.Sprintf("Image Layers :\n"))
+		buffer.WriteString(fmt.Sprintf("%s", treeData))
+
+	}
+	return &buffer, nil
+}
+
+// Builds a Map of layerID to LayerInfo ( adds links Parent layer and child layers for each layer.)
+func buildImageLayersDependencyMap(ctx context.Context, i *Image, whatrequires bool) (map[string]*LayerInfo, error) {
+	imgs, _ := i.imageruntime.store.Images()
+	layers, _ := i.imageruntime.store.Layers()
```

Comment:
The `Image` parameter can be a `Runtime` instead, `Image` is only used for the `image runtime` value, and conceptually the input image is unnecessary and confusing.

Reply:
Still outstanding.